bundles/org.openhab.binding.sunsynk/src/main/java/org/openhab/binding/sunsynk/internal/SunSynkInverter.java

```diff
@@ -0,0 +1,275 @@
+    private RealTimeInData realTimeDataIn; // realtime data
+
+    public SunSynkInverter(SunSynkInverterConfig config) {
+        // this.gsn = config.getgsn();
+        // this.access_token = config.getToken();
+        this.sn = config.getsn();
+        this.refresh = config.getRefresh();
+        this.alias = config.getAlias();
+    }
+
+    public String sendGetState(Boolean batterySettingsUpdate) { // Entry method to class for updating internal state
+                                                                // from the inverter
+        logger.debug("Will get STATE for Inverter {} serial {}", this.alias, this.sn);
+        // Get inverter infos
+        String response = null;
+        try {
+            if (batterySettingsUpdate == null) {
+                batterySettingsUpdate = false;
+            }
+            if (!batterySettingsUpdate) { // normally get settings to track changes made by other UIs
+                logger.debug("Trying Common Settings");
+                response = getCommonSettings(); // battery charge settings
+            }
+            logger.debug("Trying Grid Real Time Settings");
+            response = getGridRealTime(); // grid status
+            logger.debug("Trying Battery Real Time Settings");
+            response = getBatteryRealTime(); // battery status
+            logger.debug("Trying Temperature History");
+            response = getInverterACDCTemperatures(); // get Inverter temperatures
+            logger.debug("Trying Real Time Solar");
+            response = getRealTimeIn(); // may not need this one used for solar values could use Inverter (but would
+                                        // loose Solar power now)
+        } catch (Exception e) {
+            logger.debug("Failed to get Inverter API information: {} ", e.getMessage());
+            int found = e.getMessage().indexOf("Authentication challenge without WWW-Authenticate header");
+            if (found > -1) {
+                return "Authentication Fail";
+            }
+            return "Failed";
+        }
+        logger.debug("Successfully got and parsed new data for Inverter {} serial {}", this.alias, this.sn);
+        return response;
+    }
+
+    // ------ GETTERS ------ //
+    public Settings getBatteryChargeSettings() {
+        return this.batterySettings;
+    }
+
+    public Battery getRealTimeBatteryState() {
+        return this.realTimeBattery;
+    }
+
+    public Grid getRealTimeGridStatus() {
+        return this.grid;
+    }
+
+    public Daytemps getInverterTemperatureHistory() {
+        return this.inverter_day_temperatures;
+    }
+
+    public RealTimeInData getRealtimeSolarStatus() {
+        return this.realTimeDataIn;
+    }
+
+    // ------ SETTERS ------ //
+
+    // ------ RETRIEVERS ------//
+    // ------ Battery charge settings ------ //
+    // https://api.sunsynk.net/api/v1/common/setting/2211229948/read
+    @SuppressWarnings("null")
```

Comment:
Try to prevent supression, or if you really can't work around it, use inline suppresion to have the scope as narrow as it can be. Can be applied to other parts of the binding too.

Reply:
Okay  done